TrailBlaze/Networking/PaceImprovementTracking.h

```diff
@@ -7,22 +7,26 @@
 
 #import <Foundation/Foundation.h>
 #import "CoreLocation/CoreLocation.h"
+#import "CoreMotion/CoreMotion.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
+
 @interface PaceImprovementTracking : NSObject
+@property (strong, nonatomic) CMPedometer  *pedometer;
+@property (strong, nonatomic) NSMutableArray *currentPacesArray;
+@property (strong, nonatomic) NSArray *bestPacesArray;
+@property (strong, nonatomic) NSArray *polylinePoints;
```

Comment:
do all these need to be public and defined in the header? they're also publicly mutable since you didn't specify `readonly`

a practice I like to keep is to make public APIs as minimal as possible - this includes both properties and functions. only exposing what's necessary prevents misuse and give you some guarantees about how a component will behave at runtime.

Reply:
Since PaceImprovementTracker is not a VC there isn't an @interface in the m file to declare properties, and setting them as i-vars in the @implementation would not allow for the use of "self" to access the properties. Is there another way to make them private?

Reply:
if you define them as iVars in the .m file you can access them like this in a block:

```
self->_someInstanceVariable
```

you could also just add an interface declaration in the .m file with properties. this makes them private to the implementation file. like so:

```
@interface PaceImprovementTracking ()

@property BOOL _someFlag;

@end
```